### PR TITLE
Adding templates for R Server for HDInsight

### PR DESCRIPTION
**Pull request by matt1883**
### Changelog
* New templates: 101-hdinsight-linux-rserver-ssh-publickey, 101-hdinsight-linux-rserver-ssh-password

### Description of the change
Added two ARM templates for deploying a new HDInsight cluster with a new Azure Storage account and R Server for HDInsight.

### CODE REVIEW

File: 101-hdinsight-rserver-ssh-password/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+		"apiVersion": "[variables('clusterApiVersion')]",
+		"dependsOn": ["[concat('Microsoft.HDInsight/clusters/',parameters('clusterName'))]"],
+		"properties": {
+			"marketPlaceIdentifier": "Microsoft.RServerForHDInsight.8.0.3",
+			"computeProfile": {
+				"roles": [{
+					"name": "edgenode",
+					"targetInstanceCount": 1,
+					"hardwareProfile": {
+						"vmSize": "Standard_D4_v2"
```
Comment (singhkay) [inline 101-hdinsight-rserver-ssh-password/azuredeploy.json]: What about other VM sizes?

File: 101-hdinsight-rserver-ssh-password/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+					"container": "[parameters('clusterName')]",
+					"key": "[listKeys(resourceId('Microsoft.Storage/storageAccounts', parameters('clusterStorageAccountName')), variables('defaultApiVersion')).key1]"
+				}]
+			},
+			"computeProfile": {
+				"roles": [{
+					"name": "headnode",
+					"targetInstanceCount": "2",
+					"hardwareProfile": {
+						"vmSize": "Standard_D3_v2"
```
Comment (singhkay) [inline 101-hdinsight-rserver-ssh-password/azuredeploy.json]: Other vm sizes?

Comment (matt1883) [inline 101-hdinsight-rserver-ssh-password/azuredeploy.json]: Not for this version -- there will be a future improvement to HDInsight templates to have enums for node sizes.

File: 101-hdinsight-rserver-ssh-password/azuredeploy.json (hunk 1 above)
Comment (matt1883) [inline 101-hdinsight-rserver-ssh-password/azuredeploy.json]: Not for this version -- there will be a future improvement to HDInsight templates to have enums for node sizes.

**Comment by singhkay** [comment]
lgtm, merged

**Pull request merged**